Enterprise-CMCS/macpro-mdct-mfp
# One default target population must be applicable

Deployed url: https://d3krpqx3uqolmw.cloudfront.net/

### Description

Expected behavior: If all default target population drawers are complete and they are all marked 'No' (not applicable) then there is an error message indicating at least one must be applicable, and the page is considered incomplete on Review and Submit.

Added conditional logic to cypress tests to ensure we are marking target populations as applicable. Shoutout to Nick for writing an incredible cypress tests framework and shoutout to Ajaita for helping me use it properly!

I also removed all our `data-cy` logic in favor of better matchers 😁 

### Related ticket(s)


CMDCT-3488

---
### How to test

- Create a WP
- Go to Transition Benchmarks and select 'No' for all of the four default populations
- Verify that an error pops up on the page and the Review and Submit indicates that page as incomplete
- Change any number of populations to 'Yes' and verify the error goes away and the page is complete
- Verify that adding more pops, toggling defaults back and forth, etc. all work as expected

---
### Pre-review checklist


---
### Pre-merge checklist


#### Review

---
### Feedback requested
Please advise if you think this could be better! These were the best locations I could find to make the updates.
Testing this in the ModalDrawerReportPage is really complex to set up. Open to ideas.
